src/components/SelectionList/RadioListItem.tsx

```diff
@@ -17,30 +22,56 @@ function RadioListItem({
     onDismissError,
     shouldPreventDefaultFocusOnSelectRow,
     rightHandSideComponent,
-    checkmarkPosition,
     isMultilineSupported = false,
 }: RadioListItemProps) {
     const styles = useThemeStyles();
+    const theme = useTheme();
     const StyleUtils = useStyleUtils();
 
+    const handleCheckboxPress = useCallback(() => {
+        if (onCheckboxPress) {
+            onCheckboxPress(item);
+        } else {
+            onSelectRow(item);
+        }
+    }, [item, onCheckboxPress, onSelectRow]);
+
     return (
         <BaseListItem
             item={item}
             wrapperStyle={[styles.flex1, styles.justifyContentBetween, styles.sidebarLinkInner, styles.userSelectNone, styles.optionRow, isFocused && styles.sidebarLinkActive]}
-            selectMultipleStyle={[StyleUtils.getCheckboxContainerStyle(20), StyleUtils.getMultiselectListStyles(!!item.isSelected, !!item.isDisabled)]}
             isFocused={isFocused}
             isDisabled={isDisabled}
             showTooltip={showTooltip}
             canSelectMultiple={canSelectMultiple}
             onSelectRow={onSelectRow}
-            onCheckboxPress={onCheckboxPress}
             onDismissError={onDismissError}
             shouldPreventDefaultFocusOnSelectRow={shouldPreventDefaultFocusOnSelectRow}
             rightHandSideComponent={rightHandSideComponent}
-            checkmarkPosition={checkmarkPosition}
             keyForList={item.keyForList}
         >
             <>
+                {canSelectMultiple && (
+                    <PressableWithFeedback
+                        accessibilityLabel={item.text ?? ''}
+                        role={CONST.ROLE.BUTTON}
+                        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                        disabled={isDisabled || item.isDisabledCheckbox}
+                        onPress={handleCheckboxPress}
+                        style={[styles.cursorUnset, StyleUtils.getCheckboxPressableStyle(), item.isDisabledCheckbox && styles.cursorDisabled]}
+                    >
+                        <View style={[StyleUtils.getCheckboxContainerStyle(20), StyleUtils.getMultiselectListStyles(!!item.isSelected, !!item.isDisabled)]}>
+                            {item.isSelected && (
+                                <Icon
+                                    src={Expensicons.Checkmark}
+                                    fill={theme.textLight}
+                                    height={14}
+                                    width={14}
+                                />
+                            )}
+                        </View>
+                    </PressableWithFeedback>
+                )}
```

Comment:
I think for `RadioListItem` the above added block is not necessary - in this type of list item we don't use multi-selection, and the checkmark that indicate selection is handled by the `item.rightElement` property

Reply:
actually it is maybe worth considering replacing this `item.rightElement` usage in favor of implementing this selection indicator in the `RadioListItem` component itself

Reply:
Ok, my bad - but still, this is something that could be handled by the specific list item component instead coupling it inside the `BaseListItem` component. Can we refactor this too?